syntax = "proto3";

service DocumentService {
  // client services

  rpc InsertCommand (Insert) returns (Response) {}

  rpc DeleteCommand (Delete) returns (Response) {}

  // server services

  rpc SendInsert (InsertParams) returns (Response) {}

  rpc SendDelete (DeleteParams) returns (Response) {}
}

// client messages

message Char {
  string char = 1;
}

message Response {
  string message = 1;
}


// server messages

message Insert {
  int32 index = 1;
  string char = 2;
}

message InsertParams {
  int32 index = 1;
  string char = 2;
  int32 server_id = 3;
  bool tumbstamp = 4;
  repeated int32 timestamp = 5;
  string replica_id = 6;
}

message Delete {
  int32 index = 1;
}

message DeleteParams {
  int32 index = 1;
  int32 server_id = 2;
  bool tumbstamp = 3;
  repeated int32 timestamp = 4;
  string replica_id = 5;
}